### ISSUE TITLE
redux/ngrx support

### ISSUE THREAD

**Issue by JonnyBGod** [mean-expert-official/loopback-sdk-builder#1]
What are your thoughts on including generators for both redux since you want to support react and ngrx for angular2?

I am not sure if we should include these in this project or spin another project specific for it. What do you think?

**Comment by jonathan-casarrubias** [comment]
Hey @JonnyBGod thanks for the interest in collaborating on this project.

So, my plan is to provide support for other frameworks so react and ngrx are totally in.

Do you think you can collaborate with these?, also I have a friend who loves react so I think he will like to help on that.

thoughts?

**Comment by JonnyBGod** [comment]
I am interested in ngrx + angular2.

I have a fair experience with angular2 now but just starting with ngrx/store. My plan now is to try out your angular2 generator and from there manually convert it to use ngrx/store.

Once I feel confident with it will work on the generator. Meanwhile, will hep you tweak all angular2 related bits.

**Comment by jonathan-casarrubias** [comment]
Sounds good.. I have not moved that much the angular2... I have been working hard on the nativescript2 generator, I will be implementing the equivalent functionality for the angular2 for web asap

**Comment by JonnyBGod** [comment]
Another thought...

I am not a fan of having a huge file with all interfaces and services. I really at some point we should try to separate everything and generate helper index files. This way it will be easy to import just the parts you need.

**Comment by jonathan-casarrubias** [comment]
100% Agree on that, the big lb-services file has always been annoying I do vote for creating different modules bundled together instead one big file

**Comment by jonathan-casarrubias** [comment]
@JonnyBGod I pushed my latests changes... at this moment both (nativescript2 and angular2) have exactly the same functionality.

I created a shared folder with the templates that can be shared, but I needed to modify many of these.

I see you have created the ngrx folder, so you will need to port whatever is on shared that works for that generator, and leave the specific ones inside the ngrx folder.

Right now the index.js is exactly the same for angular2 and nativescript2.. I think we could use only 1 but I think the ngrx will be different because I see you have created many different files, so maybe only the ngrx version would be to be different? thoughts?

Regards,
Jon
